### ISSUE TITLE
[bug] match function with populate 

### ISSUE THREAD

**Issue by kareemhamdy100** [Automattic/mongoose#10391]
http://thecodebarbarian.com/mongoose-5-5-static-hooks-and-populate-match-functions.html#populate-match-function
the example that happened with me when use something like example in the link and

```
post = await BlogPost.findOne().populate({
path: 'comments',
// If match is a function, it receives the blog post doc as a parameter
match: doc => ({
authorId: doc.authorId,
blog: doc._id,
deleted: { $ne: true } })
});

```

but instead using findOne I used find to return a list with data  from other collection and I found out it's create wrong query object and the expected result is that every blog returns
with it's comments only
but what happened it's returns every comment related to the author despite of the is that comment for the corresponding blog or not

because the created query become something like this

comments.find( { deleted : {$ne : true} ,
$or: [
{authorId : <blog1.authorId> , blog : <blog1._Id> },
{authorId : <blog2.authorId> , blog : <blog2._Id>}
]
}

so if authorId is the same the condition of or will be correct

**Comment by vkarpov15** [comment]
@kareemhamdy100 are the results incorrect? Because that sounds like expected behavior: using a `match` function uses a `$or` under the hood and then filters the results in memory using http://npmjs.com/package/sift.

**Comment by kareemhamdy100** [comment]
@vkarpov15  yes it's produce  incorrect result I will try to get more clear example
